File: chapter2.1/caesar/caesar.c
/**
 *
 * Tahmidul Alam
 * CS50
 * Ms. Tanzosh
 *
 * Caesar Cypher
*/

#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar 13\n");
        return 1;
    }

    int input = atoi(argv[1]);

    if (input < 0)
    {
        printf("Usage: ./caesar 13\n");
        return 1;
    }
    else
    {
        string input_code = get_string("Enter Text: ");
        printf("ciphertext: ");
        for (int i = 0, n = strlen(input_code); i < n; i++)
        {
            if (islower(input_code[i]))
            {
                printf("%c", (((input_code[i] + input) - 'a') % 26) + 'a');
            }
            else if (isupper(input_code[i]))
            {
                printf("%c", (((input_code[i] + input) - 'A') % 26) + 'A');
            }
            else
            {
                printf("%c", input_code[i]);
            }
        }
    }
    printf("\n");
    return 0;
}

File: 2018-19/unit6/caesar/caesar.py
'''
    Tahmidul Alam
    CS50AP
    Caesar
    caesar.py
'''

import sys
import cs50


def caesar(inp, cipher_code):
    # Print out the pre-text
    print("ciphertext: ", end="")

    # Counter for iteration
    c = 0
    # For each character in string inp
    for i in inp:
        # Check lowercase
        if inp[c].islower():
            # Encryption formula (Put the math in numbers)
            lower_characters_num = (((ord(inp[c]) + cipher_code) - ord('a')) % 26) + ord('a')
            # Encryption formula generated (the result is an ASCII code; put it in the corresponding character)
            lower_characters = chr(lower_characters_num)
            print("{}".format(lower_characters), end="")
        # Check uppercase
        elif inp[c].isupper():
            # Encryption formula (Put the math in numbers)
            upper_characters_num = (((ord(inp[c]) + cipher_code) - ord('A')) % 26) + ord('A')
            # Encryption formula generated (the result is an ASCII code; put it in the corresponding character)
            upper_characters = chr(upper_characters_num)
            # Print out the character
            print("{}".format(upper_characters), end="")
        else:
            # The text may have spaces or punctuations
            print(inp[c], end="")
        # Counter increase to access the next index
        c += 1
    # Print a new line
    print()


def main():
    # Number of command line inputs
    # Only 2 is permitted
    if len(sys.argv) != 2:
        print("Usage: python caesar.py 13")
        exit(1)

    # Assuming that CLI passed, get the cipher code that we want to encrypt
    cipher_code = int(sys.argv[1])

    # The cipher code must be positive
    if (cipher_code < 0):
        print("Usage: python caesar.py 13")
        exit(1)
    # Cipher code test pass
    else:
        # Pre-text
        print("Enter Text: ", end="")
        # Ask the user for the text they want to encrypt
        input_code = cs50.get_string()
        # Launch the encryption
        caesar(input_code, cipher_code)


# Check for any unnecessary errors
if __name__ == "__main__":
    main()